sdk/python/lib/test/test_next_serialize.py

```diff
@@ -295,10 +295,33 @@ async def test_output_all(self):
 
         other = Output.from_input(99)
         combined = Output.all(out, other)
+        combined_dict = Output.all(out=out, other=other)
         deps = []
         prop = await rpc.serialize_property(combined, deps)
-        self.assertListEqual(deps, [res])
+        prop_dict = await rpc.serialize_property(combined_dict, deps)
+        self.assertSetEqual(set(deps), {res})
         self.assertEqual([42, 99], prop)
+        self.assertEqual({"out": 42, "other": 99}, prop_dict)
+
+    @pulumi_test
+    async def test_output_all_failure(self):
+        self.assertRaises(ValueError, Output.all)
+
+    @pulumi_test
+    async def test_output_all_args_prioritized_if_mixed(self):
+        res = FakeCustomResource("some-resource")
+        fut = asyncio.Future()
+        fut.set_result(42)
+        known_fut = asyncio.Future()
+        known_fut.set_result(True)
+        out = Output({res}, fut, known_fut)
+
+        other = Output.from_input(99)
+        combined = Output.all(out, other=other)
+        deps = []
+        prop = await rpc.serialize_property(combined, deps)
+        self.assertSetEqual(set(deps), {res})
+        self.assertEqual([42], prop)
```

Comment:
I've written it this way because this is how the current `Output.all()` function behaves. However, I would prefer that we error in the case of mixed inputs and clarify that either args or kwargs should be used but not both. That would technically be a breaking change though.

Reply:
> However, I would prefer that we error in the case of mixed inputs and clarify that either args or kwargs should be used but not both. That would technically be a breaking change though.

I'm inclined to take this very minor breaking change as part of this change.  It's not clear that code that was taking advantage of this before could possibly have been doing what the user expected it to do.  (And more likely, there is no code out there taking advantage of this).